src/components/atoms/ButtonNG/ButtonNG.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { CSSProperties } from "react";
+import { Link } from "react-router-dom";
+import classNames from "classnames";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { faCircleNotch } from "@fortawesome/free-solid-svg-icons/faCircleNotch";
+
+import "./ButtonNG.scss";
+
+export type ButtonGradientType =
+  | "gradient"
+  | "admin-gradient"
+  | "danger-gradient";
+
+export type ButtonType = "button" | "reset" | "submit";
+export type ButtonVariant = "primary" | "secondary" | "white" | "dark";
+export type ButtonIconSize = "1x" | "2x" | "3x";
+
+export interface ButtonProps {
+  className?: string;
+  style?: CSSProperties;
+  disabled?: boolean;
+  onClick?: () => void;
+  isLink?: boolean;
+  linkTo?: string;
+  newTab?: boolean;
+  loading?: boolean;
+  gradient?: ButtonGradientType;
+  type?: ButtonType;
+  variant?: ButtonVariant;
+  iconOnly?: boolean;
+  iconName?: IconProp;
+  iconSize?: ButtonIconSize;
+}
+
+export const ButtonNG: React.FC<ButtonProps> = ({
+  children,
+  className = "",
```

Comment:
please name it more clearly, like `containerClassName`

Reply:
Why is `containerClassName` clear?

Reply:
discussed and left as a `className`